src/wasm.rs

```diff
@@ -0,0 +1,334 @@
+    }
+
+    fn section_data_by_name(&self, section_name: &str) -> Option<Cow<'data, [u8]>> {
+        match section_name {
+            // Known wasm section names.
+            "Type" => self.module
+                .type_section()
+                .and_then(|s| serialize_to_cow(s.clone())),
+            "Import" => self.module
+                .import_section()
+                .and_then(|s| serialize_to_cow(s.clone())),
+            "Function" => self.module
+                .function_section()
+                .and_then(|s| serialize_to_cow(s.clone())),
+            "Table" => self.module
+                .table_section()
+                .and_then(|s| serialize_to_cow(s.clone())),
+            "Memory" => self.module
+                .memory_section()
+                .and_then(|s| serialize_to_cow(s.clone())),
+            "Global" => self.module
+                .global_section()
+                .and_then(|s| serialize_to_cow(s.clone())),
+            "Export" => self.module
+                .export_section()
+                .and_then(|s| serialize_to_cow(s.clone())),
+            "Start" => self.module
+                .start_section()
+                .and_then(|s| serialize_to_cow(elements::VarUint32::from(s))),
+            "Element" => self.module
+                .elements_section()
+                .and_then(|s| serialize_to_cow(s.clone())),
+            "Code" => self.module
+                .code_section()
+                .and_then(|s| serialize_to_cow(s.clone())),
+            "Data" => self.module
+                .data_section()
+                .and_then(|s| serialize_to_cow(s.clone())),
+            // Custom sections.
+            _ => {
+                for s in self.module.sections() {
+                    if let elements::Section::Custom(ref c) = *s {
+                        if c.name() == section_name {
+                            return Some(Cow::from(c.payload().to_owned()));
+                        }
+                    }
+                }
+                None
+            }
+        }
+    }
+
+    fn sections(&'file self) -> Self::SectionIterator {
+        WasmSectionIterator {
+            sections: self.module.sections().iter(),
+            _phantom: PhantomData,
+        }
+    }
+
+    fn symbols(&'file self) -> Self::SymbolIterator {
+        WasmSymbolIterator {
+            file: self,
+            _phantom: PhantomData,
+        }
+    }
+
+    fn dynamic_symbols(&'file self) -> Self::SymbolIterator {
+        WasmSymbolIterator {
+            file: self,
+            _phantom: PhantomData,
+        }
+    }
+
+    fn symbol_map(&self) -> SymbolMap<'data> {
+        SymbolMap {
+            symbols: Vec::new(),
+        }
+    }
+
+    fn is_little_endian(&self) -> bool {
+        true
+    }
+
+    fn has_debug_symbols(&self) -> bool {
+        // We ignore the "name" section, and use this to mean whether the wasm
+        // has DWARF.
```

Comment:
I don't understand this comment. This function is doing what I would expect. Perhaps this function is poorly named, would `has_debug_info` be better?

Reply:
Wasm has the concept of a name section that provides names for functions, which would otherwise be called `wasm-function[42]`, and in the context of wasm this is usually considered "having debug symbols".

I think that `has_debug_info` would be a better name, but it isn't too important.